In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats

# Settings 
plt.style.use('seaborn-v0_8-whitegrid')
pd.set_option('display.max_columns', None)



In [4]:
df = pd.read_csv('../data/raw/ee.csv')
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 768 entries, 0 to 767
Data columns (total 10 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   relative_compactness  768 non-null    float64
 1   surface_area          768 non-null    float64
 2   wall_area             768 non-null    float64
 3   roof_area             768 non-null    float64
 4   overall_height        768 non-null    float64
 5   orientation           768 non-null    int64  
 6   glazing_area          768 non-null    float64
 7   glazing_area_dist     768 non-null    int64  
 8   heating_load          768 non-null    float64
 9   cooling_load          768 non-null    float64
dtypes: float64(8), int64(2)
memory usage: 60.1 KB


# 1. Load Data (10 pts)
#### How many rows does your raw dataset have?
768.
#### What column is your target variable?
heating_load and cooling_load, although I think using just one might be fine as from our correlations observed they were incredibly similar.
#### What issues from your Notebook 01 will you address in this notebook?
Aforementioned collinearity.

# 2. Drop Rows with Missing TARGET (15 pts)
#### How many rows had a missing target value? 
Zero.
#### What percentage of the dataset does that represent?
0.0%
#### How many rows remain after dropping?
768.

In [5]:
# Showing the lack of missing values in the dataset
df.isnull().sum()

relative_compactness    0
surface_area            0
wall_area               0
roof_area               0
overall_height          0
orientation             0
glazing_area            0
glazing_area_dist       0
heating_load            0
cooling_load            0
dtype: int64

# 3. Simple Cleaning (10 pts)
#### Note what was removed (duplicate rows, data-leakage columns w/ explanation, impossible values)

# 4 Train-Test Split (10 pts)
#### Print shapes of X_train, X_test, y_train, y_test
#### How many rows in training and test sets?


# 5. Handle Missing Values (25 pts)
There are none to begin with, thankfully.

# 6. Handle Outliers (30 pts)


# Key Findings Summary
#### Finding 1
#### Finding 2
#### Finding 3
#### Finding 4